HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -153,4 +159,9 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
     # New cards for DT local reco
     process = customiseFor34612(process)
 
+    # GEM
+    from Configuration.ProcessModifiers.dd4hep_cff import dd4hep
+    if menuType in ["GRun","HIon","PIon","PRef"]:
+        addCustomiseForRun3GEMGeometry34788_ = (~dd4hep).makeProcessModifier( customiseForRun3GEMGeometry34788 )
+
```

Comment:
I'm not very familiar with the syntax of `makeProcessModifier`, but I don't see how this is modifying the `process` (assuming that is the intention). I would have expected
```
(~dd4hep).makeProcessModifier(customiseForRun3GEMGeometry34788).apply(process)
```
Also, I was curious to know if the check on `menuType` is necessary. If not, it's maybe better to remove it.

Reply:
It would also be best to move the `import` call outside the function.

Reply:
AFAIK, `makeProcessModifier` acts on the process as a whole, so we don't need `process` again. 

For the menu, as I test, it will not work with the Fake menu, i.e. Run1 or Run2 config. I assume because GEM does not exist. This is to provide a temporary solution until GEM unpacker will integrate with CSC (I hope soon), then the whole of this can be removed.

I can move import away, but then it will need to be at the beginning, or just before the function. It is fine to me.

Reply:
> AFAIK, `makeProcessModifier` acts on the process as a whole, so we don't need `process` again.

That is true for cfi/cff fragments that get load into the `process` via `process.load()` (which is the stage where `cms.Process` calls the `apply()`). In this case the `ProcessModifier` is created in a customize function, in which case the `ProcessModifier` does not automatically propagate to the `process`, and therefore, as @missirol pointed out, the `ProcessModifier.apply()` needs to be called explicitly.

Reply:
Thanks @makortel 
I've updated the PR.

Reply:
Actually, what is the purpose/meaning of the tilde ```~``` in front of dd4hep?

Reply:
If not dd4hep. When DD4hep is used, I will get
```
two EventSetup Producers want to deliver type="GEMGeometry" label=""
 from record MuonGeometryRecord. The two providers are 
1) type="GEMGeometryESModule" label=""
2) type="GEMGeometryESModule" label="gemGeometry"
```

Reply:
It's a "not": if the dd4hep modifier is _not_ active, then call customiseFor34788

Reply:
That will be a pain when we move the ES module into the HLT menus in ConfDb, as that is not dd4hep aware. Why this different treatment dd4hep vs non-dd4hep?

Reply:
I guess that when`--procModifier dd4hep` is used, it calls this. But I have to check with Geometry.
Apart from GEM ES module, why we don't see issues from other ES modules (in HLT menu in ConfDB) that should be called in the same way.